pkg/logging/logging.go

```diff
@@ -170,16 +170,24 @@ func loggingProcessAdapter(ctx context.Context, driver Driver, dataStore string,
 	processLogFunc := func(reader io.Reader, dataChan chan string) {
 		defer wg.Done()
 		defer close(dataChan)
-		scanner := bufio.NewScanner(reader)
-		for scanner.Scan() {
-			if scanner.Err() != nil {
-				log.L.Errorf("failed to read log: %v", scanner.Err())
+		scanner := bufio.NewReader(reader)
+
+		for {
+			s, err := scanner.ReadString('\n')
```

Comment:
My bet is that `bufio.Scanner.Scan` does NOT include LF, while `ReadString` does.
Pretty sure it will trip tests. If I am right, just strip a possibly trailing LF before sending?

Reply:
@CerberusQc read here if you do not know yet how to run specific integration tests locally:
https://github.com/containerd/nerdctl/blob/main/docs/testing.md

Reply:
@CerberusQc read here if you do not know yet how to run specific integration tests locally:
https://github.com/containerd/nerdctl/blob/main/docs/testing.md

Reply:
wanted to keep it DRY but would need to process the TrimSuffix even if there is an error that is not `io.EOF`.
I am really open to idea on this one

Reply:
On that note:

"
ReadString reads until the first occurrence of delim in the input, returning a string containing the data up to and including the delimiter. If ReadString encounters an error before finding a delimiter, it returns the data read before the error and the error itself (often io.EOF). ReadString returns err != nil if and only if the returned data does not end in delim.
"
So:
- err == nil is the only case where you should trim
- you should send the data regardless of what error we get

I would maybe go for something like this (untested...):

```go
	if err == nil {
		s = strings.TrimSuffix(s, "\n")
	}else if err != nil && err != io.EOF {
		log.L.WithError(err).Error("failed to read log")
	}
	dataChan <- s
```

Reply:
We will need to exit the loop when an error occur, but 
```GO
	s, err := scanner.ReadString('\n')
	if err != nil && err != io.EOF {
		log.L.WithError(err).Error("failed to read log")
		return
	}
	dataChan <- strings.TrimSuffix(s, "\n")
```
should do the job since the 2 cases we want to send is when there is no error or when the error is not EOF right?

Reply:
I think we should send in all cases - there may be data, with or without error (and regardless of the error).

take what I say with salt here - not my forte

Reply:
how would we exit the for loop in that case? Maybe I am missing something with channel and defer.

Reply:
I would break.

Here is something that should be closer to something functioning:

```go
processLogFunc := func (reader io.Reader, dataChan chan string) {
	defer wg.Done()
	defer close(dataChan)
	scanner := bufio.NewReader(reader)

	for {
		s, err := scanner.ReadString('\n')
		if err == nil {
			s = strings.TrimSuffix(s, '\n')
		}
		dataChan <- s
		if err != nil && err != io.EOF {
			log.L.WithError(err).Error("failed to read log")
			break
		}
	}
}

```

Alternatively, stuff like that would work too to break the loop:

```
var err error
for ; err == nil; {

}
```